plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestBooleanCoercer.java

```diff
@@ -53,11 +53,80 @@ public void testBooleanToLowerBoundedVarchar()
                 .hasMessageContaining("Varchar representation of false exceeds varchar(1) bounds");
     }
 
+    @Test
+    public void testVarcharToBoolean()
+    {
+        // Valid false values
+        assertVarcharToBooleanCoercion("FALSE", false);
+        assertVarcharToBooleanCoercion("OFF", false);
+        assertVarcharToBooleanCoercion("NO", false);
+        assertVarcharToBooleanCoercion("0", false);
+        assertVarcharToBooleanCoercion("", false);
+
+        // false values in mixed cases
+        assertVarcharToBooleanCoercion("FAlSE", false);
+        assertVarcharToBooleanCoercion("OfF", false);
+        assertVarcharToBooleanCoercion("nO", false);
+        assertVarcharToBooleanCoercion("no", false);
+
+        // True values
+        assertVarcharToBooleanCoercion("YES", true);
+        assertVarcharToBooleanCoercion("TRUE", true);
+        assertVarcharToBooleanCoercion("TR", true);
+        assertVarcharToBooleanCoercion("T", true);
+        assertVarcharToBooleanCoercion("Y", true);
+        assertVarcharToBooleanCoercion("1", true);
```

Comment:
add test for `0` in ORC as well

Reply:
I think we have added it in `testVarcharToBooleanForOrc`